### PR TITLE
Update required-review config

### PR DESCRIPTION
**Pull request by anomiex**
## Proposed changes:

This makes the following changes:

* Taking advantage of the new `consume` option to DRY things out.
* Allow everyone to update `composer.lock` and `changelog/*` in all projects, not only plugins.
* Fix #28359, as written the "Monorepo itself" rule still applied to docs so only jetpack-approvers were actually allowed. This is done by leaving `docs/**` in the rule with `consume` set an moving "Monorepo itself" after.
* The "Jetpack and packages" rule now only covers Jetpack. A second rule covers those packages (and now js-packages too) that are maintained by Ground Control. All other packages are left to team rules or the catch-all at the end, so we don't have to exclude them anymore.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* **Do not merge until after a new version of the required-review package containing #29317 has been released!**
* Use the test.sh script to run against various PRs, I guess.

### CODE REVIEW

**Comment by kraftbj** [comment]
I think the list is capitalization sensitive, but afaik, GH user names are unique without regard to capitalization. While you're in there, is that something you could look at? https://github.com/Automattic/jetpack/pull/29320

**Comment by anomiex** [comment]
> I think the list is capitalization sensitive, but afaik, GH user names are unique without regard to capitalization. While you're in there, is that something you could look at? #29320

Sure, but not on this PR.

File: .github/files/required-review.yaml [hunk 1]
```diff
@@ -25,38 +18,43 @@
...
+   - 'projects/js-packages/i18n-loader-webpack-plugin/**'
+   - 'projects/js-packages/remove-asset-webpack-plugin/**'
+   - 'projects/js-packages/webpack-config/**'
+   - 'projects/packages/autoloader/**'
+   - 'projects/packages/changelogger/**'
+   - 'projects/packages/codesniffer/**'
+   - 'projects/packages/ignorefile/**'
+   - 'projects/packages/phpcs-filter/**'
+  teams:
+   - jetpack-approvers
```
Comment (jeherve) [inline .github/files/required-review.yaml:51]: I do not fully understand that part. Given that the Approvers group already is the fallback, why do we need to add those specific use-cases?

Comment (anomiex) [reply .github/files/required-review.yaml:51]: We don't need to, at least not now. OTOH, this clearly documents that these packages are specifically intended to be owned, and keeps them that way should we decide to change the default for some reason.

**Comment by anomiex** [comment]
This will need reconsideration after #29426. Possibly we won't need anything further here, but maybe some of the rearrangement will still make sense.

**Pull request closed without merging**